Common.mk

```diff
@@ -3,91 +3,156 @@ MAKEFLAGS+=--no-builtin-rules --warn-undefined-variables
 .SHELLFLAGS:=-eu -o pipefail -c
 .SUFFIXES:
 
-RELEASE_BRANCH?=
 RELEASE_ENVIRONMENT?=development
-ARTIFACT_BUCKET?=my-s3-bucket
-GIT_HASH=$(shell git rev-parse HEAD)
+GIT_HASH?=$(shell git rev-parse HEAD)
 
+COMPONENT=$(REPO_OWNER)/$(REPO)
+MAKE_ROOT=$(BASE_DIRECTORY)/projects/$(COMPONENT)
+BUILD_LIB=${BASE_DIRECTORY}/build/lib
+OUTPUT_BIN_DIR?=$(OUTPUT_DIR)/bin/$(REPO)
+
+#################### AWS ###########################
 AWS_REGION?=us-west-2
 AWS_ACCOUNT_ID?=$(shell aws sts get-caller-identity --query Account --output text)
+ARTIFACT_BUCKET?=my-s3-bucket
+IMAGE_REPO?=$(if $(AWS_ACCOUNT_ID),$(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com,localhost:5000)
+####################################################
+
+#################### LATEST TAG ####################
 BRANCH_NAME?=main
 LATEST_TAG?=latest
 ifneq ("$(BRANCH_NAME)","main")
 	LATEST_TAG=$(BRANCH_NAME)
 endif
+####################################################
 
-COMPONENT=$(REPO_OWNER)/$(REPO)
+#################### CODEBUILD #####################
 ifdef CODEBUILD_SRC_DIR
 	ARTIFACTS_PATH?=$(CODEBUILD_SRC_DIR)/$(PROJECT_PATH)/$(CODEBUILD_BUILD_NUMBER)-$(CODEBUILD_RESOLVED_SOURCE_VERSION)/artifacts
 	CLONE_URL=https://git-codecommit.$(AWS_REGION).amazonaws.com/v1/repos/$(REPO_OWNER).$(REPO)
 else
 	ARTIFACTS_PATH?=$(MAKE_ROOT)/_output/tar
-	CLONE_URL?=https://github.com/$(COMPONENT).git	
+	CLONE_URL=https://github.com/$(COMPONENT).git	
 endif
+####################################################
 
-MAKE_ROOT=$(BASE_DIRECTORY)/projects/$(COMPONENT)
-OUTPUT_DIR?=_output
-OUTPUT_BIN_DIR?=$(OUTPUT_DIR)/bin/$(REPO)
-
-BUILD_LIB=${MAKE_ROOT}/../../../build/lib
+#################### GIT ###########################
+GIT_CHECKOUT_TARGET?=$(REPO)/eks-anywhere-checkout-$(GIT_TAG)
+GIT_PATCH_TARGET?=$(REPO)/eks-anywhere-patched
+####################################################
 
-ATTRIBUTION_TARGET=$(wildcard ATTRIBUTION.txt) $(wildcard $(RELEASE_BRANCH)/ATTRIBUTION.txt)
+#################### RELEASE BRANCHES ##############
+HAS_RELEASE_BRANCHES?=false
+RELEASE_BRANCH?=
+SUPPORTED_K8S_VERSIONS=$(shell yq e 'keys | .[]' $(BASE_DIRECTORY)/projects/kubernetes-sigs/image-builder/BOTTLEROCKET_OVA_RELEASES)
+ifneq ($(RELEASE_BRANCH),)
```

Comment:
Can we also override the LATEST_IMAGE to have a tag that includes a GIT_TAG corresponding to the release branch? For cloud-provider-vsphere, we had `$(GIT_TAG)-$(LATEST_TAG)` which made the image `...manager:v1.20.0-latest` and `...manager:v1.21.0-latest`. Without that, both would be [`...manager:latest`](https://prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp.s3.us-west-2.amazonaws.com/pr-logs/pull/aws_eks-anywhere-build-tooling/158/cloud-provider-vsphere-tooling-presubmit/1458571256222715904/build-container-build-log.txt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAUTLHWUYSE4667FR5%2F20211111%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20211111T052128Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEFUaCXVzLXdlc3QtMiJHMEUCIQCKJ3IBWBt7f0p6iNgIFDrq%2BfnzSiQs8XC3eKSCIgQo8AIgIz6jnK8AhBgVYog63bYATVVU4IG3iH4vm8NwQp7Htq8q%2BQQI%2Fv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARABGgwzMTY0MzQ0NTgxNDgiDGRFaKUhtMaaeTXpaCrNBJbPRhMAjqnPiPdEHAHOg3%2B8wTiLC5BojpchLsrQ2S34cVuMaOhV5apm07OdkVHKt%2FDeg6tgVXjYZ2LUknpnSJUDr5jtrJk2vAFzyxBKizsGMvRfU9G54w%2BXBaFSxTaw%2BnStBxYiXDt2ueoaeG2%2FIavg8W8M1a4AQtljAVwMM1Crt0k6mNXGzZgih2NjUWIV5Sa8Uv3IQRCS4%2BWhNZOvMgZpp2mTltvZMMHhwp3B0aMtHbTCdCit%2BuOSez10Fnjmasw0TOY3bZOAQyraQEHxxDnovRlIXgMbVTSwOzZZP8m7qXG0cEGtB4vTtAlZZUeMzvrdSkiADb8pz6PiTZiUBiC3lojUzrwgAU6zSxO6fnu2fd7ReLqtoUXg3D%2B9VvB6yU3tDRqkncpfiarf2m7qUwdPNNseWQYgVZzWJwamKGJe08ByEY2QCOdol4VEGFz8gGslY7qvrKXHi3FCkRYOYiWia5fgwMMU1curZDGiVq2E%2Bt%2Frw4C1UVmRA3YzLeBRri%2BFZE%2FVrx6JxMY6OFuvd%2FQ4E%2FD7p4Edcxd2WKP%2FpWVcEzBKrULJzVYD8j6eEfpS%2FCTIcsce5EVJtFTQrgoQ9ZaMyeevnqMiSYGvPrG6nxjCC945alHtDkhIVWCkQD5vBPxS7Eb5D1F6IiSJMBMbIqF1c%2BwItFxVBJCPQAm8bUd2U3LaNlBOCBfI83iThm6qmeOgcK9qOJcUCJuzx1cS4hsyH5dUuhNsviQCxIngDRdlmD2hXfL1nVt8c49LbcYxriZFjADM73DPs%2F9OpLAwn8CyjAY6mgFSGi3KJsMJUwoiZOpyQKmfckJtXIVSE2%2FWURc6rb67kUbqU%2FWocnP%2FNBqTwGdHqNCd6ccCCNeU%2FBFmUriWNWvBhT9vlMiW3%2FOgK9VGJlX8bwEx29BtxiYBlsFbUzCHfZKOgjy0%2FlFFlK2ZxAKWzuQF7Ec0K12DeBd9pGjJkS74LCY4CO3PuPD0%2FNkcMofAoY6ju7qb%2FQeBgXL8&X-Amz-SignedHeaders=host&X-Amz-Signature=50a300a6bf107da84567268da2355d6e52710f051f232438e20a9da48da9bcec), so the first one would get overwritten.

Reply:
great catch!

Reply:
fixed